### PR TITLE
v1.1.5

### PR DESCRIPTION
**Pull request by drtriumph**
- Updating Firebase dependency to latest 2.0.x
- `firebase bootstrap` and `firebase deploy` no longer rely on temporary files / folders
- Updated node module dependencies

### CODE REVIEW

File: lib/app.js [hunk 1]
```diff
@@ -300,6 +297,16 @@ function getSettings(argv) {
   return settings;
 }
 
+function mkDirs(filePath) {
+  filePathParts = filePath.split(path.sep);
```
Comment (jwngr) [inline lib/app.js]: Missing `var`.

File: lib/upload.js [hunk 2]
```diff
@@ -61,52 +67,36 @@ module.exports = {
...
-      });
-      var form = r.form();
-      form.append('site', readStream);
+    var r = request({
+      url: url,
+      method: 'DELETE',
+      json: true
+    }, function(err, response, body) {
+      var failed = (err || !body || !body.success);
+      setTimeout(callback, 0, failed, body ? body.directory : undefined);
```
Comment (jwngr) [inline lib/upload.js]: `body ? body.directory : undefined` can be simplified to `body && body.directory`

File: lib/api.js [hunk 3]
```diff
@@ -55,14 +55,13 @@ var api = {
...
       method: 'PUT',
-      json: true,
       body: rules
     }, function(err, response, body) {
       if (err) {
         console.log('SERVER ERROR'.red + ' - Please try again');
         process.exit(1);
       }
-      setTimeout(callback, 0, response.statusCode, body);
+      setTimeout(callback, 0, response.statusCode, body ? JSON.parse(body) : body);
```
Comment (jwngr) [inline lib/api.js]: Can be simplified to `body && JSON.parse(body)`

File: lib/app.js [hunk 4]
```diff
@@ -689,28 +699,12 @@ module.exports = {
                 .child('hosting/versions')
                 .child(settings.firebase)
                 .push();
-        var bar = null;
-        var total = 0;
         directoryRef.on('value', function(snapshot) {
           var status = snapshot.child('status').val();
-          if (status === 'deploying') {
```
Comment (jwngr) [inline lib/app.js]: I see you removed the code for the "deploying" stage without replacing it with anything. Is this no longer needed or something?

**Comment by jwngr** [comment]
I made a few comments, but I really don't know this code that well. Things looked good in general but we will need to make sure we've thoroughly tested it before we release it.

File: lib/app.js (hunk 4 above)
Comment (drtriumph) [inline lib/app.js]: Correct, the new delete endpoint won't ever go through the 'deploying' state, instead straight to 'removed' state, and so we don't need the progress bar

Comment (jwngr) [inline lib/app.js]: Ahh okay that makes total sense.

**Pull request merged**